src/Combinatorics/EnumerativeCombinatorics/partitions.jl

```diff
@@ -123,6 +125,39 @@ function getindex_safe(P::Partition{T}, i::IntegerUnion) where T
   return (i > length(data(P)) ? zero(T) : getindex(data(P), Int(i)))
 end
 
+base(P::Partitions) = P.n
+
+Base.eltype(::Partitions{T}) where T = Partition{T}
+
+function Base.show(io::IO, ::MIME"text/plain", P::Partitions)
+  if is_terse(io)
+    print(io, "Iterator")
```

Comment:
```suggestion
    print(io, "Iterator over partitions")
```
Actually this function is a case where I am tempted to say the terse version makes no sense, the regular oneline version is "terse enough" in all real world examples.


Wonder what @simonbrandhorst @thofma @fieker think

Reply:
This should be kept consistent with the iterators for Compostions and WeakCompositions (in any case od decision)

Reply:
Yeah, I didn't know how short/terse a `terse` version is supposed to be in the other iterators.

Reply:
yeah, I copied this from (Weak)Compositions for consistency, but I do think we should make them both just a little less terse.

Reply:
Triage agrees with @fingolfin

Reply:
@mjrodgers can you apply the same printing change as in [`376e9ec` (#3389)](https://github.com/oscar-system/Oscar.jl/pull/3389/commits/376e9ec1b506e6466834866d68a5a6dbd68b658a) for Compositions and WeakCompositions as well?

Reply:
sure, done